ssetests/tests_basic_parsing.go

```diff
@@ -1,19 +1,67 @@
 package ssetests
 
 import (
+	"crypto/rand"
+	"encoding/base64"
 	"fmt"
 	"time"
 
 	"github.com/launchdarkly/sse-contract-tests/framework/ldtest"
+
+	"github.com/stretchr/testify/assert"
 )
 
+func generateRandomString(length int) string {
+	b := make([]byte, length)
+	_, err := rand.Read(b)
+	if err != nil {
+		panic(err)
+	}
+	return base64.StdEncoding.EncodeToString(b)
+}
+
 func DoBasicParsingTests(t *ldtest.T) {
 	t.Run("one-line message in one chunk", func(t *ldtest.T) {
 		_, stream, client := NewStreamAndSSEClient(t)
 		stream.Send("data: Hello\n\n")
 		client.RequireSpecificEvents(t, EventMessage{Data: "Hello"})
 	})
 
+	t.Run("two messages spanning 3 chunks with shared chunk", func(t *ldtest.T) {
+		// This test primarily tests situations where the implementation over allocates buffers to decrease the total
+		// number of buffers. This test helps ensure that the used size in the buffer is properly tracked.
+		_, stream, client := NewStreamAndSSEClient(t)
+		stream.Send("data: test")
+		stream.Send("test\n\ndata:")
```

Comment:
It's.. unfortunate we don't have a streaming interface for a thing called stream (unless we do?).

 Otherwise we could generate the random data on demand and keep memory usage minimal.